kind/README.adoc

```diff
@@ -0,0 +1,204 @@
+on a laptop. It currently supports the following features: 
+
+* Create/delete a K8s cluster using the latest cluster version, or the one you specify, 
+* Launch a local container registry exposed on port `localhost:5000`, 
+* Set up an Ingress
+controller to route the traffic. 
+
+The bash script has been developed supporting the installation of a docker registry according to the following options: 
+
+* Unsecure: plain HTTP
+* Secured: HTTP/HTTPS endpoints available
+
+== Prerequisite
+
+To be able to install Kind using this project the following tools are required.
+
+* jq: https://stedolan.github.io/jq/download/
+* Kind: https://github.com/kubernetes-sigs/kind/releases
+* Docker desktop: https://www.docker.com/products/docker-desktop
+* openssl
+
+== Usage
+
+From the repository root execute the following command.
+
+[source,bash]
+----
+./kind/deploy.sh command [parameters,...]
+----
+
+The available commands are:
+
+.Script commands
+[%header,cols="2m,4"]
+|===
+| Command | Meaning
+
+| install | Install the kind cluster
+| remove | Remove the kind cluster
+|===
+
+=== Relevant options
+
+Other relevant parameters are the following.
+
+.Script commands
+[%header,cols="2,5"]
+|===
+| Command | Meaning
+
+| `--ingress`
+
+[.fuchsia]#string# / [.red]#required# 
+a| Ingress to be deployed. 
+
+Choices: 
+
+* nginx
+* kourier
+
+| `--cluster-name`
+
+[.fuchsia]#string#
+a| Name of the cluster.
+
+Default: kind
+
+| `--port-map`
+
+[.fuchsia]#string#
+a| Comma separated list of ports to map on kind config.
+
+_e.g._: `ContainerPort1:HostPort1,...`
+
+[WARNING]
+====
+The ports required to be accessed from outside the kind cluster must be mapped between the host and the kind container using this parameter.
+====
+
+| `--provider`
+
+[.fuchsia]#string#
+a| Allows selecting the container runtime to be used. 
+
+Choices: 
+
+* *`docker` <= Default*
+* `podman`
+
+|===
+
+
+== Deploy
+
+Deploying a kind cluster is as simple as executing the following command.
+
+.kind deploy statement
+[source,bash]
+----
+curl -s -L "https://raw.githubusercontent.com/snowdrop/k8s-infra/main/kind/deploy.sh" | bash -s install --ingress nginx --cluster-name twm --provider podman --port-map "30000:30000,31000:31000"
+----
+
+=== Post deployment steps
+
+Allow using an insecure container registry.
+
+Add the `kind-registry.conf` file to the `/etc/containers/registries.conf.d/` folder having the following contents.
+
+[source]
+----
+[[registry]]
+location = "localhost:5000"
+insecure = true
+----
+
+== Undeploy
+
+To undeploying a previousy deployed kind cluster use the `remove` action on the kind deploy script.
+
+.kind undeploy statement
+[source,bash]
+----
+curl -s -L "https://raw.githubusercontent.com/snowdrop/k8s-infra/main/kind/deploy.sh" | bash -s remove --cluster-name twm
+----
+
+== Secured docker registry
+
+To deploy a local secured (using htpasswd) docker registry use the `--secure-registry` parameter.
```

Comment:
Can we do that independently to install the docker registry as HTTP/HTTPS = TLS registry ?

Reply:
I'm not sure if this is possible as the local registry is included in the kind cluster `yaml` by the `[plugins."io.containerd.grpc.v1.cri".registry.mirrors."localhost:${REGISTRY_PORT}"]
    endpoint = ["http://${registry_name}:${REGISTRY_PORT}"]` directive. I'd have to check if we can install the cluster and then patch the cluster definition by changing this parameter. 

Should I create a new issue to address this or you think not having this is a blocker for rolling out the script as is?

@cmoulliard

Reply:
No.